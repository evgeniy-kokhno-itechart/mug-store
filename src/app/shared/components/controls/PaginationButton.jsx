import React, { Component } from 'react';
import { PropTypes } from 'prop-types';

class PaginationButton extends Component {
  handleChange = () => {
    const { page, handlePageChange } = this.props;
    handlePageChange(page);
  };

  render() {
    const { page } = this.props;
    return (
      <button type="button" className="page-link clickable" onClick={this.handleChange}>
        {page}
      </button>
    );
  }
}

PaginationButton.propTypes = {
  page: PropTypes.number.isRequired,
  handlePageChange: PropTypes.func.isRequired,
};

export default PaginationButton;
