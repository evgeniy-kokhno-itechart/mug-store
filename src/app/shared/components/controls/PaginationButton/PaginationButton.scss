@import "Variables";

.pagination-button {
    font: inherit;
    padding: $padding-vertical;
    min-width: 2.3rem;
    background-color: transparent;
    color: inherit;
    border-color: inherit;
}