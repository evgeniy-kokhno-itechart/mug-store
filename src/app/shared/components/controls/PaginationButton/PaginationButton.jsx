import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import './PaginationButton.scss';

class PaginationButton extends Component {
  handleChange = () => {
    const { page, handlePageChange } = this.props;
    handlePageChange(page);
  };

  render() {
    const { page } = this.props;
    return (
      <button type="button" className="pagination-button focusable clickable" onClick={this.handleChange}>
        {page}
      </button>
    );
  }
}

PaginationButton.propTypes = {
  page: PropTypes.number.isRequired,
  handlePageChange: PropTypes.func.isRequired,
};

export default PaginationButton;
