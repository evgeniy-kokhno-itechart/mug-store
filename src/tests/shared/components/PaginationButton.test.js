/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PaginationButton from '../../../app/shared/components/controls/PaginationButton/PaginationButton';

configure({ adapter: new Adapter() });

describe('<PaginationButton />', () => {
  let PaginationButtonWrapper;
  let handlePageChange;

  beforeEach(() => {
    handlePageChange = jest.fn();
    PaginationButtonWrapper = shallow(<PaginationButton page={1} handlePageChange={handlePageChange} />);
  });

  it('check render properly', () => {
    expect(PaginationButtonWrapper).toMatchSnapshot();
  });

  it('check onClick passes proper arguments to handlePageChange', () => {
    PaginationButtonWrapper.find('button').simulate('click');
    expect(handlePageChange).toHaveBeenCalledWith(1);
  });
});
