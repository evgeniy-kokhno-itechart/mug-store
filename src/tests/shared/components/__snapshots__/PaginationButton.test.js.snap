// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationButton /> check render properly 1`] = `
<button
  className="pagination-button focusable clickable"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
